extensions/pkg/controller/healthcheck/general/managed_resource.go

```diff
@@ -79,10 +79,17 @@ func (healthChecker *ManagedResourceHealthChecker) Check(ctx context.Context, re
 	}
 	if isHealthy, err := managedResourceIsHealthy(mcmDeployment); !isHealthy {
 		healthChecker.logger.Error(err, "Health check failed")
+
+		var errorCodes []gardencorev1beta1.ErrorCode
+		configurationProblemRegexp := regexp.MustCompile(`(?i)(error during apply of object .* is invalid:)`)
```

Comment:
Nit: you could move this inside the variable declaration above or use it inline as well.